core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -139,11 +144,46 @@ static __maybe_unused const char *shres2str_state(enum stm32mp_shres id)
 	return shres2str_state_tbl[id];
 }
 
+/* GPIOZ bank pin count depends on SoC variants */
+#ifdef CFG_DT
+/* A light count routine for unpaged context to not depend on DTB support */
+static int gpioz_nbpin = -1;
+
+static unsigned int get_gpioz_nbpin_unpg(void)
+{
+	if (gpioz_nbpin < 0)
+		panic();
+
+	return (unsigned int)gpioz_nbpin;
+}
+KEEP_PAGER(get_gpioz_nbpin_unpg);
+
 static unsigned int get_gpioz_nbpin(void)
+{
+	if (gpioz_nbpin < 0) {
```

Comment:
I would have preferred one function to read out and save the pin count and another to return it. That way it's clear when the driver is initializing, instead of being in a maybe half initialized state until `stm32mp1_init_shres()` has been run.